nni/retiarii/execution/cgo_engine.py

```diff
@@ -79,12 +81,15 @@ def register_graph_listener(self, listener: AbstractGraphListener) -> None:
         self._listeners.append(listener)
 
     def _send_trial_callback(self, paramater: dict) -> None:
-        for listener in self._listeners:
-            listener.on_resource_used(0)  # FIXME: find the real resource id
+        if self.resources <= 0:
+            _logger.warning('There is no available resource, but trial is submitted.')
+        print(paramater)
+        self.resources -= paramater['training_kwargs']['n_model']
```

Comment:
so for now you have not considered the case that multiple model are packed on one GPU?

Reply:
see your comments